mathics/core/pattern.py

```diff
@@ -289,27 +289,28 @@ def per_expr(yield_expr, expr_groups, sum=0):
 
                         if expr_groups:
                             expr, count = expr_groups.popitem()
-                            max_per_pattern = count / len(patterns)
+                            max_per_pattern = int(count / len(patterns))
```

Comment:
so this is interesting. I had yield_expr throw right away because "get wrapping" was called with the wrong type of argument ... a sequence rather than a dict. How did you figure out that you needed to call "per_name" again with the yield_func (which is "yield_expr"?)  I didn't see that 🥇

Reply:
Multiple reasons:

- It was [the original intention](https://github.com/rebcabin/Mathics/pull/1/files#diff-aff3cf11e8ba9d512e83515618b3e82dL311);
- The broken code of `yield_expr` was a left-over from some previous (or unachieved future) state of things, but had the desired pseudo-logic.

The next is mixed with speculation:

- `get_wrappings` would return an iterator over or a sequence of wrappings, then for each wrapping the further code would call `per_name` with reducing groups (from the left);
- `get_wrappings` itself for `Orderless` [goes over a sequence of permutations](https://github.com/sealemar/Mathics/blob/52ebb0c02cf14792ee48e8e0c21a2a4b8dd1f785/mathics/core/pattern.py#L347), the sequence being a [collection of unique arguments](https://github.com/rebcabin/Mathics/pull/1/files#diff-aff3cf11e8ba9d512e83515618b3e82dR291) along with the number of their occurrences from the original expression;
- Thus the idea is to go over all groups and all expressions in each of the groups.